service/frontend/workflowHandler.go

```diff
@@ -4546,6 +4549,17 @@ func (wh *WorkflowHandler) cleanScheduleSearchAttributes(searchAttributes *commo
 	return searchAttributes
 }
 
+// This may mutate saPtr and *saPtr
+func (wh *WorkflowHandler) addSearchAttribute(saPtr **commonpb.SearchAttributes, key string, value *commonpb.Payload) {
```

Comment:
Is there a helper function for this already? I didn't see one. It should probably be moved somewhere

Reply:
I would move it to https://github.com/temporalio/temporal/blob/master/common/searchattribute/search_attirbute.go#L73.